src/main/java/appeng/client/render/cablebus/FacadeBuilder.java

```diff
@@ -19,370 +19,364 @@
+			.build()//
+	);
+	private ThreadLocal<Quad> collectors = ThreadLocal.withInitial( Quad::new );
+
+	public void buildFacadeQuads( BlockRenderLayer layer, CableBusRenderState renderState, long rand, List<BakedQuad> quads, Function<ResourceLocation, IBakedModel> modelLookup )
 	{
+		BakedPipeline pipeline = pipelines.get();
+		Quad collectorQuad = collectors.get();
+		boolean transparent = AEApi.instance().partHelper().getCableRenderMode().transparentFacades;
+		Map<EnumFacing, FacadeRenderState> facadeStates = renderState.getFacades();
+		List<AxisAlignedBB> partBoxes = renderState.getBoundingBoxes();
+		Set<EnumFacing> sidesWithParts = renderState.getAttachments().keySet();
+		IBlockAccess parentWorld = renderState.getWorld();
+		BlockPos pos = renderState.getPos();
+		BlockColors blockColors = Minecraft.getMinecraft().getBlockColors();
 		boolean thinFacades = isUseThinFacades( partBoxes );
 
-		CubeBuilder builder = new CubeBuilder( this.format, quads );
-
-		facadesState.forEach( ( side, state ) ->
+		for( Entry<EnumFacing, FacadeRenderState> entry : facadeStates.entrySet() )
 		{
-			AxisAlignedBB facadeBox = getFacadeBox( side, thinFacades );
-			AEAxisAlignedBB cutOutBox = getCutOutBox( facadeBox, partBoxes );
+			EnumFacing side = entry.getKey();
+			int sideIndex = side.ordinal();
+			FacadeRenderState facadeRenderState = entry.getValue();
 			boolean renderStilt = !sidesWithParts.contains( side );
-
-			try
+			if( layer == BlockRenderLayer.CUTOUT && renderStilt )
 			{
-				this.addFacade( layer, facadesState, side, cutOutBox, thinFacades, renderStilt, rand, builder );
+				for( ResourceLocation part : PartCableAnchor.FACADE_MODELS.getModels() )
+				{
+					IBakedModel partModel = modelLookup.apply( part );
+					QuadRotator rotator = new QuadRotator();
+					quads.addAll( rotator.rotateQuads( gatherQuads( partModel, null, rand ), side, EnumFacing.UP ) );
+				}
 			}
-			catch( Throwable t )
+			//If we are forcing transparency and this isn't the Translucent layer.
+			if( transparent && layer != BlockRenderLayer.TRANSLUCENT )
 			{
-				AELog.debug( t );
+				continue;
 			}
-		} );
-	}
-
-	public static TextureAtlasAndTint getSprite( IBakedModel blockModel, IBlockState state, EnumFacing facing, long rand )
-	{
 
-		TextureAtlasAndTint firstFound = null;
-		BlockRenderLayer orgLayer = MinecraftForgeClient.getRenderLayer();
-
-		try
-		{
-			// Some other mods also distinguish between layers, so we're doing this in a loop from most likely to least
-			// likely
-			for( BlockRenderLayer layer : BlockRenderLayer.values() )
+			IBlockState blockState = facadeRenderState.getSourceBlock();
+			//If we aren't forcing transparency let the block decide if it should render.
+			if( !transparent )
 			{
-
-				ForgeHooksClient.setRenderLayer( layer );
-
-				for( BakedQuad bakedQuad : blockModel.getQuads( state, facing, rand ) )
+				if( !blockState.getBlock().canRenderInLayer( blockState, layer ) )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1066)